pandas/core/arrays/period.py

```diff
@@ -228,6 +303,183 @@ def start_time(self):
     def end_time(self):
         return self.to_timestamp(how='end')
 
+    def __repr__(self):
+        return '<{}>\n{}\nLength: {}, dtype: {}'.format(
+            self.__class__.__name__,
+            [str(s) for s in self],
+            len(self),
+            self.dtype
+        )
+
+    def __len__(self):
```

Comment:
is the super definition ok here instead?

Reply:
`DatetimeLikeArrayMixin` has it indeed, so that can be removed here